build.ps1

```diff
@@ -155,13 +159,19 @@ if (-not ($repoPath))
 ###
 # Picks up msbuild from vs2019 installation
 ###
-$VsRequirements = @(
+$VsRequirements = [System.Collections.ArrayList]@(
     'Microsoft.Component.MSBuild'
     'Microsoft.VisualStudio.Workload.NativeDesktop'
     'Microsoft.VisualStudio.Component.VC.14.29.16.11.ATL.Spectre'
```

Comment:
Any chance we could point these at less specific version numbers? That way if you have something later installed it doesn't fail.

Reply:
VS component ids are quite annoying. These ids are the only ids for v142 toolset in VS2022 (which require versioning). 

If we use the "latest toolset" id, then it's v142 for VS2019 and v143 for VS2022. I could update VSwhere to check different ids for VS versions. That work can be done in a separate PR.